File: Modulo06/HerancaNamespace/Coordenador.cpp
#include "Coordenador.h"

namespace poo {

    Coordenador::Coordenador(string nome, int CPF, double salario, string curso) :
    Professor(nome, CPF, salario), curso(curso) {
    }

    double Coordenador::getSalario() const {
        return Professor::getSalario() + 2000;
    }
    
    string Coordenador::getCurso() const {
        return curso;
    }

    void Coordenador::imprime() const {
        Professor::imprime();
        cout << "Curso: " << getCurso() << endl;
    }

}

File: Modulo12/Professor-Java/br/ufscar/dc/poo/Coordenador.java
package br.ufscar.dc.poo;

public class Coordenador extends Professor {

    private String curso;

    public Coordenador(String nome, int CPF, double salario, String curso) {
        super(nome, CPF, salario);
        this.curso = curso;
    }

    public double getSalario() {
        return super.getSalario() + 2000;
    }

    public String getCurso() {
        return curso;
    }

    public void imprime() {
        super.imprime();
        System.out.println("Curso: " + curso);
    }
}
